src/commonMethods/commonFunctions.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * This function activates a given plugin
+ * @param plugin_name
+ * @returns current state of given plugin
+ */
+import constants from './constants'
+
+export const pluginActivate = function(plugin_name) {
+  return this.$thunder.api.Controller.activate({ callsign: plugin_name })
+    .then(() =>
+      this.$thunder.api.call(constants.controllerPlugin, 'status').then(
+        result =>
+          result.filter(p => {
+            return p.callsign === plugin_name
+          })[0].state
+      )
+    )
+    .catch(err => err)
+}
+
+/**
+ * This function deactivates a given plugin
+ * @param plugin_name
+ * @returns current state of given plugin
+ */
+export const pluginDeactivate = function(plugin_name) {
+  return this.$thunder.api.Controller.deactivate({ callsign: plugin_name })
+    .then(() =>
+      this.$thunder.api.call(constants.controllerPlugin, 'status').then(
+        result =>
+          result.filter(p => {
+            return p.callsign === plugin_name
+          })[0].state
+      )
+    )
+    .catch(err => err)
+}
+
+/**
+ * This function resumes or suspends WebKitBrowser plugin
+ * @param action
+ */
+export const webKitBrowserActions = function(action) {
+  return this.$thunder.api.WebKitBrowser.state(action)
+    .then(() => {
+      return this.$thunder.api.call(constants.controllerPlugin, 'status').then(
+        result =>
+          result.filter(p => {
+            return p.callsign === 'WebKitBrowser'
+          })[0].state
+      )
+    })
+    .catch(err => err)
+}
+
+/**
+ * This function sets the URL
+ * @param URL
+ * @returns URL
+ */
+export const settingUrl = function(URL) {
+  return this.$thunder.api.WebKitBrowser.url(URL)
```

Comment:
I'd call this setWebKitURL, as its limited to the WebKitBrowser instance

Reply:
Updated the code as per the comment